pkg/sql/alter_index_visible.go

```diff
@@ -28,9 +33,49 @@ type alterIndexVisibleNode struct {
 func (p *planner) AlterIndexVisible(
 	ctx context.Context, n *tree.AlterIndexVisible,
 ) (planNode, error) {
-	return nil, unimplemented.Newf(
-		"Not Visible Index",
-		"altering an index to visible or not visible is not supported yet")
+	if err := checkSchemaChangeEnabled(
+		ctx,
+		p.ExecCfg(),
+		"ALTER INDEX VISIBILITY",
+	); err != nil {
+		return nil, err
+	}
+
+	// Check if the table actually exists. expandMutableIndexName returns the
+	// underlying table. If the index name is specified with a table name, it
+	// checks if the table exists. Otherwise, it searches the table using the
+	// index name. If no table is found, 1. IfExists is specified, tableDesc
+	// returned is nil. 2. No IfExists, error is returned.
+	_, tableDesc, err := expandMutableIndexName(ctx, p, &n.Index, !n.IfExists /* requireTable */)
+	if err != nil {
+		// Error if no table is found and IfExists is false.
```

Comment:
nit: these kinds of comments add little value but increase clutter, remove them.

Same comment applies below.